charts/internal/calico/templates/kube-controller/deployment-calico-kube-controllers.yaml

```
@@ -34,7 +34,7 @@ spec:
         gardener.cloud/role: system-component
     spec:
       nodeSelector:
-        beta.kubernetes.io/os: linux
+{{ toYaml .Values.nodeSelector | indent 8 }}
```

```suggestion
      {{- if .Values.nodeSelector }}
      nodeSelector:
{{ toYaml .Values.nodeSelector | indent 8 }}
      {{- end }}
```

You can omit the node selector if it is not set in the values.

---

Done